apps/studio/components/interfaces/Reports/v2/ReportChartV2.tsx

```diff
@@ -0,0 +1,186 @@
+import { useQuery } from '@tanstack/react-query'
+
+import Panel from 'components/ui/Panel'
+import { useFillTimeseriesSorted } from 'hooks/analytics/useFillTimeseriesSorted'
+import { useCurrentOrgPlan } from 'hooks/misc/useCurrentOrgPlan'
+import { useSelectedOrganizationQuery } from 'hooks/misc/useSelectedOrganization'
+import { cn } from 'ui'
+import type { AnalyticsInterval } from 'data/analytics/constants'
+import type { ReportConfig } from 'data/reports/v2/reports.types'
+import ComposedChart from 'components/ui/Charts/ComposedChart'
+import { ReportChartUpsell } from './ReportChartUpsell'
+import { useState } from 'react'
+
+export interface ReportChartV2Props {
+  report: ReportConfig
+  projectRef: string
+  startDate: string
+  endDate: string
+  interval: AnalyticsInterval
+  updateDateRange: (from: string, to: string) => void
+  functionIds?: string[]
+  edgeFnIdToName?: (id: string) => string | undefined
+  isLoading?: boolean
+  className?: string
+  syncId?: string
+}
+
+export const ReportChartV2 = ({
+  report,
+  projectRef,
+  startDate,
+  endDate,
+  interval,
+  updateDateRange,
+  functionIds,
+  edgeFnIdToName,
+  isLoading,
+  className,
+  syncId,
+}: ReportChartV2Props) => {
+  const { data: org } = useSelectedOrganizationQuery()
+  const { plan: orgPlan } = useCurrentOrgPlan()
+  const orgPlanId = orgPlan?.id
+
+  const isAvailable =
+    report.availableIn === undefined || (orgPlanId && report.availableIn.includes(orgPlanId))
+
+  const canFetch = orgPlanId !== undefined
+
+  const {
+    data: queryResult,
+    isLoading: isLoadingChart,
+    error,
+  } = useQuery(
+    ['report-v2', report.id, projectRef, startDate, endDate, interval, functionIds],
```

Comment:
this is just a nit and not sure if it'll be cleaner

for RQ keys we typically start with the project ref first, and wrap configuration related params into a single object
so e.g `['projects', projectRef, 'report-v2', report.id, { startData, endDate, interval, functionIds }]`